src/components/ItemList.svelte

```diff
@@ -59,12 +59,35 @@
   }
 
   function fullTextSearch(query, searchItems) {
+    let filteredSearchItems = searchItems;
+    let searchQuery = query;
+
+    if (query.startsWith("tags:") || query.startsWith("origin:")) {
+      const labelType = query.split(":")[0]; // e.g. "tags" or "origin"
+      // separate label query from search query
+      const [label, searchValue] = query.split(":")[1].split(" ");
+
+      filteredSearchItems = searchItems.filter(
+        (item) => item[labelType] && item[labelType].includes(label)
+      );
+
+      // returns all items of that label if no extra search term is specified
+      if (!searchValue) return filteredSearchItems;
+      searchQuery = searchValue;
+    }
+
     const results = [
-      ...new Set(searchIndex.search(query).flatMap((match) => match.result)),
+      ...new Set(
+        searchIndex
+          .search(searchQuery, { limit: 10000 })
```

Comment:
we need to manually set the limit here to get back _all_ results because FlexSearch only returns 100 entries: https://github.com/nextapps-de/flexsearch#limit--offset

Reply:
Can you add a comment about this?